partiql-spi/src/main/java/org/partiql/spi/catalog/PartiQLSystemCatalog.java

```diff
@@ -0,0 +1,66 @@
+package org.partiql.spi.catalog;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.spi.function.Aggregation;
+import org.partiql.spi.function.Builtins;
+import org.partiql.spi.function.Function;
+
+import java.util.Collection;
+
+/**
+ * <p>
+ * This implements the PartiQL System Catalog.
+ * </p>
+ * <p>
+ * It provides the implementation for the PartiQL System Catalog, which is a built-in catalog
+ * that provides access to the PartiQL language and its built-in functions and aggregations.
+ * </p>
+ * @see Session.Builder
+ */
+final class PartiQLSystemCatalog implements Catalog {
+
+    /**
+     * TODO
+     */
+    @NotNull
+    private final String name;
+
+    /**
+     * Creates a new PartiQL System Catalog with the given name.
+     * @param name the name of the PartiQL System Catalog
+     */
+    PartiQLSystemCatalog(@NotNull String name) {
+        this.name = name;
+    }
```

Comment:
Can't make a protected top-level class. Updated the Javadocs to make the visibility more apparent, since there is no package-private modifier. And added the singleton. See latest commit: https://github.com/partiql/partiql-lang-kotlin/pull/1670/commits/d1b33056d3405fdd4ee1f8dd23d20aa1144e3dd9

If we want to expose this class in the future, sure. But we don't need to right now.